activerecord/lib/active_record/querying.rb

```diff
@@ -40,7 +40,7 @@ module Querying
     def find_by_sql(sql, binds = [], preparable: nil, &block)
       result_set = connection.select_all(sanitize_sql(sql), "#{name} Load", binds, preparable: preparable)
       column_types = result_set.column_types.dup
-      columns_hash.each_key { |k| column_types.delete k }
+      columns_hash.merge(attributes_to_define_after_schema_loads).each_key { |k| column_types.delete k }
```

Comment:
It seems that it is better to always use the `connection.schema_cache.columns_hash` here. `attributes_to_define_after_schema_loads` can contain attributes that are not in the database and we use this just to remove from `column_types`